pvlib/iotools/tmy.py

```diff
@@ -198,9 +214,16 @@ def read_tmy3(filename, coerce_year=None, recolumn=True):
     # NOTE: as of pvlib-0.6.3, min req is pandas-0.18.1, so pd.to_timedelta
     # unit must be in (D,h,m,s,ms,us,ns), but pandas>=0.24 allows unit='hour'
     data.index = data_ymd + pd.to_timedelta(shifted_hour, unit='h')
-
-    if recolumn:
+    if map_variables is None:
         data = _recolumn(data)  # rename to standard column names
```

Comment:
We should still only apply the ``recolumn()`` function if the parameter ``recolumn`` is True.

```suggestion
        if recolumn:
            data = _recolumn(data)  # rename to standard column names
```